connect/runtime/src/test/java/org/apache/kafka/connect/storage/KafkaConfigBackingStoreTest.java

```diff
@@ -751,6 +764,21 @@ public void testPutTaskConfigsDoesNotResolveAllInconsistencies() throws Exceptio
         PowerMock.verifyAll();
     }
 
+    @Test(expected = ConfigException.class)
+    public void testExceptionOnStartWhenConfigTopicHasMultiplePartitions() throws Exception {
+        expectConfigure();
+        expectStart(Collections.emptyList(), Collections.emptyMap());
+
+        expectGetPartitionCount(2);
+
+        PowerMock.replayAll();
+
+        configStorage.setupAndCreateKafkaBasedLog(TOPIC, DEFAULT_DISTRIBUTED_CONFIG);
+        configStorage.start();
```

Comment:
The test looks great! 
But fyi, we have `org.junit.Assert.assertThrows;` and `org.hamcrest.MatcherAssert.assertThat`, `org.hamcrest.CoreMatchers.containsString` in our disposal now, that allow for better assertions in tests that throw exceptions (added the full classnames because they are conflicts with deprecated methods in `junit`)